src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,205 @@
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllQuery = "SELECT c.id AS car_id, model, m.id "
+                + "AS manufacturer_id, m.name, m.country"
+                + " FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id"
+                + " WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllStatement = connection.prepareStatement(getAllQuery)) {
+            ResultSet resultSet = getAllStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCarWithManufacturerFromResultSet(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars from DB.", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET model = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStatement
+                        = connection.prepareStatement(updateQuery)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getId());
+            updateStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update " + car + " in DB.", e);
+        }
+        deleteAllRelationsBetweenCarAndDriversIfExists(car);
+        insertDrivers(car);
+        return car;
+    }
+
+    private void deleteAllRelationsBetweenCarAndDriversIfExists(Car car) {
+        String deleteRelationsQuery = "DELETE cars_drivers.* FROM cars_drivers "
+                + "WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteRelationsStatement =
+                        connection.prepareStatement(deleteRelationsQuery)) {
+            deleteRelationsStatement.setLong(1, car.getId());
+            deleteRelationsStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete relations between drivers and car "
+                    + "with id " + car.getId(), e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String softDeleteQuery = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement softDeleteStatement =
+                        connection.prepareStatement(softDeleteQuery)) {
+            softDeleteStatement.setLong(1, id);
+            return softDeleteStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }

```

Comment:
if you soft delete the car by id, what is left in the cars_drivers table? Don't we need to delete all connections between this car and drivers?